src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -123,64 +123,60 @@ private void PopulateSnippetCaches()
         {
             Debug.Assert(_expansionManager != null);
 
-            // Ideally we'd fork execution here based on whether the expansion manager is an
-            // IExpansionManager or not. Unfortunately, we cannot mention that type by name until
-            // the Roslyn build machines are upgraded to Visual Studio 2015 Update 1. We therefore
-            // need to try using IExpansionManager dynamically, from a background thread. If that
-            // fails, then we come back to the UI thread for using IVsExpansionManager instead.
-
             var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
 
-            Task.Factory.StartNew(async () => await PopulateSnippetCacheOnBackgroundWithForegroundFallback().ConfigureAwait(false),
-                            CancellationToken.None,
-                            TaskCreationOptions.None,
-                            TaskScheduler.Default).CompletesAsyncOperation(token);
+            var asyncExpansionManager = _expansionManager as IExpansionManager;
```

Comment:
What is tracking the eventual removal? In either case, a comment here explaining the otherwise seemingly "dead" code would be useful I think.